### PR TITLE
Flesh out KV support

### PR DESCRIPTION
**Pull request by mmihic**
R: @xichen2020 @robskillington

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -42,6 +50,25 @@ test-ci-unit: test-internal
 	@which goveralls > /dev/null || go get -u -f github.com/mattn/goveralls
 	goveralls -coverprofile=$(coverfile) -service=travis-ci || echo -e "\x1b[31mCoveralls failed\x1b[m"
 
+install-vendor: .gitmodules
```
Comment (xichen2020) [inline Makefile]: Should this be glide based now that we've migrated over to glide?

Comment (mmihic) [inline Makefile]: We still need submodules for uber-licence

File: kv/mock.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package kv
+
+import (
+	"github.com/golang/protobuf/proto"
+	"sync"
```
Comment (xichen2020) [inline kv/mock.go]: Reorder the import path perhaps?

File: Makefile (hunk 1 above)
Comment (mmihic) [inline Makefile]: We could move this out of vendor though so it's clear this isn't a golang dep

File: vendor/github.com/uber/uber-licence/license-fixer.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
```
Comment (robskillington) [inline vendor/github.com/uber/uber-licence/license-fixer.js]: Hm, this file is showing up as an add?  Shouldn't this be a submodule?

Comment (mmihic) [inline vendor/github.com/uber/uber-licence/license-fixer.js]: Hm, it was added as a submodule.  Let me try to re-apply in some way.

File: Makefile (hunk 1 above)
Comment (xichen2020) [inline Makefile]: ah yeah, forgot about the node module.

I'm fine either way. Someone can probably spend a day rewriting the license module in go so we don't end up having a node module in go repos :).

**Pull request merged**